### PR TITLE
SJQ2T-111: BE - Endpoint for promoting user

### PR DESCRIPTION
**Pull request by MetoMladenovski**
Implemented promote user functionality.

### CODE REVIEW

**Review by ivanfhtec — COMMENTED** [review]
seems fine

File: backend/src/main/java/com/htecgroup/skynest/controller/UserController.java [hunk 1]
```diff
@@ -427,4 +427,55 @@ public ResponseEntity<Boolean> changePassword(
...
+        @ApiResponse(
+            responseCode = "200",
+            description = "User successfully promoted",
+            content = {
+              @Content(
+                  mediaType = "application/json",
+                  schema = @Schema(implementation = String.class),
+                  examples = {@ExampleObject(value = "true")})
+            }),
+        @ApiResponse(
```
Comment (petarstoimenovskihtec) [inline backend/src/main/java/com/htecgroup/skynest/controller/UserController.java]: No 401 and 403 ?

File: backend/src/main/java/com/htecgroup/skynest/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -145,4 +146,17 @@ public void changePassword(UserChangePasswordRequest userChangePasswordRequest,
     Email email = EmailUtil.createPasswordChangeNotificationEmail(changedPasswordDto);
     emailService.send(email);
   }
+
+  @Override
+  public void promoteUser(UUID userId) {
+    UserDto userDto = findUserById(userId);
```
Comment (petarstoimenovskihtec) [inline backend/src/main/java/com/htecgroup/skynest/service/impl/UserServiceImpl.java]: I would add that in a custom annotation, maybe that PR is already merged, so you can see how to do it

File: backend/src/test/java/com/htecgroup/skynest/utils/UserDtoUtil.java [hunk 3]
```diff
@@ -28,4 +30,8 @@ public static UserDto getNotVerified() {
   public static UserDto getVerified() {
     return getNotVerified().withVerified(true).withEnabled(true);
   }
+
+  public static UserDto getNotWorker() {
```
Comment (petarstoimenovskihtec) [inline backend/src/test/java/com/htecgroup/skynest/utils/UserDtoUtil.java]: why not simple call this method to createManager or something that can be reusable in some other situation ?

File: backend/src/test/java/com/htecgroup/skynest/service/impl/UserServiceImplTest.java [hunk 4]
```diff
@@ -211,6 +211,27 @@ void editUser() {
...
+    Assertions.assertEquals(expectedErrorMessage, thrownException.getMessage());
+  }
+
+  @Test
+  void when_WorkerUser_promoteUser_ShouldPromoteUser() {
+    UserDto userDto = UserDtoUtil.getVerified();
+    doReturn(userDto).when(userService).findUserById(any());
+    UUID roleId = UUID.randomUUID();
+    when(roleService.findByName(anyString()))
+        .thenReturn(new RoleDto(roleId, RoleEntity.ROLE_MANAGER));
```
Comment (petarstoimenovskihtec) [inline backend/src/test/java/com/htecgroup/skynest/service/impl/UserServiceImplTest.java]: if you extract this role in a method you can reuse it is several places

**Comment by MetoMladenovski** [comment]
I think I've made the required changes.

File: backend/src/main/java/com/htecgroup/skynest/annotation/validator/PromoteValidator.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+public class PromoteValidator implements ConstraintValidator<AdminUserCanPromote, UUID> {
+
+  private final CurrentUserService currentUserService;
+
+  private final UserService userService;
+
+  @Override
+  public boolean isValid(UUID uuid, ConstraintValidatorContext constraintValidatorContext) {
+    String accessedUserRole = userService.getUser(uuid).getRoleName();
+    return accessedUserRole.equals(RoleEntity.ROLE_WORKER);
```
Comment (petarstoimenovskihtec) [inline backend/src/main/java/com/htecgroup/skynest/annotation/validator/PromoteValidator.java:21]: I am confused, the annotation says AdminCanPromote and here in the body a Worker role is checked? Either fix the logic if it is a mistake, or rename the method and annotation to something more descriptive

File: backend/src/test/java/com/htecgroup/skynest/service/impl/UserServiceImplTest.java [hunk 6]
```diff
@@ -215,6 +212,18 @@ void when_VerifiedAndEnabledUser_disableUser_ShouldDisableUser() {
     Assertions.assertNotNull(userEntity.getDeletedOn());
   }
 
+  @Test
+  void when_WorkerUser_promoteUser_ShouldPromoteUser() {
+    UserDto userDto = UserDtoUtil.getVerified();
+    doReturn(userDto).when(userService).findUserById(any());
+    RoleDto roleManager = RoleDtoUtil.getManagerRole();
+    when(roleService.findByName(anyString())).thenReturn(roleManager);
+    userService.promoteUser(any());
```
Comment (petarstoimenovskihtec) [inline backend/src/test/java/com/htecgroup/skynest/service/impl/UserServiceImplTest.java]: Usually in tests, when you call the actual method that you are testing, a real value is used. So pls change any() with something specific

File: backend/src/main/java/com/htecgroup/skynest/annotation/validator/PromoteValidator.java (hunk 5 above)
Comment (MetoMladenovski) [reply backend/src/main/java/com/htecgroup/skynest/annotation/validator/PromoteValidator.java:21]: Ah yes, first I was a little bit confused with the annotations and was checking admin role, then I changed it and forgot to rename it. Will do.

**Review by petarstoimenovskihtec — APPROVED** [review]

**Pull request merged**